src/main/java/com/iexec/sms/tee/session/base/SecretSessionBaseService.java

```diff
@@ -111,6 +114,46 @@ public SecretSessionBase getSecretsTokens(final TeeSessionRequest request) throw
         return sessionBase.build();
     }
 
+    /**
+     * Get tokens required for different signature-related features.
+     *
+     * @param request      Session request details
+     * @param computeStage The compute stage (pre-compute or post-compute)
+     * @return A {@code Map} containing tokens required for the signature
+     * @throws TeeSessionGenerationException if any of the required tokens is missing
+     */
+    Map<String, String> getSignTokens(final TeeSessionRequest request, final String computeStage) throws TeeSessionGenerationException {
+        final String taskId = request.getTaskDescription().getChainTaskId();
+        final String workerAddress = request.getWorkerAddress();
+        if (StringUtils.isEmpty(workerAddress)) {
+            throw new TeeSessionGenerationException(
+                    GET_SIGNATURE_TOKENS_FAILED_EMPTY_WORKER_ADDRESS,
+                    "Empty worker address - taskId: " + taskId);
+        }
+        if (StringUtils.isEmpty(request.getEnclaveChallenge())) {
+            throw new TeeSessionGenerationException(
+                    GET_SIGNATURE_TOKENS_FAILED_EMPTY_PUBLIC_ENCLAVE_CHALLENGE,
+                    "Empty public enclave challenge - taskId: " + taskId);
+        }
+        final Optional<TeeChallenge> teeChallenge = teeChallengeService.getOrCreate(taskId, true);
+        if (teeChallenge.isEmpty()) {
+            throw new TeeSessionGenerationException(
+                    GET_SIGNATURE_TOKENS_FAILED_EMPTY_TEE_CHALLENGE,
+                    "Empty TEE challenge  - taskId: " + taskId);
+        }
+        final EthereumCredentials enclaveCredentials = teeChallenge.get().getCredentials();
+        if (enclaveCredentials == null || enclaveCredentials.getPrivateKey().isEmpty()) {
+            throw new TeeSessionGenerationException(
+                    GET_SIGNATURE_TOKENS_FAILED_EMPTY_TEE_CREDENTIALS,
+                    "Empty TEE challenge credentials - taskId: " + taskId);
+        }
+        final Map<String, String> tokens = new HashMap<>();
+        tokens.put(IEXEC_TASK_ID.name(), taskId);
+        tokens.put(SIGN_WORKER_ADDRESS.name(), workerAddress);
+        tokens.put(SIGN_TEE_CHALLENGE_PRIVATE_KEY.name(), enclaveCredentials.getPrivateKey());
+        return tokens;
```

Comment:
I think we could go with something like (check indentation, mine should not be the correct one)
```suggestion
        return Map.of(
            IEXEC_TASK_ID.name(), taskId,
            SIGN_WORKER_ADDRESS.name(), workerAddress,
            SIGN_TEE_CHALLENGE_PRIVATE_KEY.name(), enclaveCredentials.getPrivateKey());
```

Reply:
Resolved in [dbfd39b](https://github.com/iExecBlockchainComputing/iexec-sms/pull/296/commits/dbfd39b146d7612e20bec9727f9de6860694d2dc)